### PR TITLE
New methods & property for PlayerAPI & EntityAPI in LLSE

### PR DESCRIPTION
**Pull request by ghost**
## Description

Please carefully read the [Contributing note](https://github.com/LiteLDev/LiteLoaderBDS/discussions/546) before making any pull requests.
And, **Do not make a pull request to merge into main branch unless it is a hotfix. Use the beta branch instead.**
## Changelog

- **EntityAPI**
    - Add new properties `canFly`, `inLava`, `inRain`, `inSnow`, `inClouds`, `isTrading`, `isRiding`.
    - Add new methods `getNameTag`, `setNameTag`, `setNameTagVisible`
- **PlayerAPI**
    - Add new properties `inLava`, `inRain`, `inSnow`, `inClouds`, `isTrading`, `isRiding`, `isGliding`.
    - Add new methods `getNameTag`, `setNameTagVisible`
    - Correct `canFly` (he return false, if the player isn't in the main world) 

## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [X] Enhancement
- [X] Documentation

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] My pull request is unique and no other pull requests have been opened for these changes
- [X] I have read the [Contributing note](https://github.com/LiteLDev/LiteLoaderBDS/discussions/546)
- [X] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: ScriptEngine/API/EntityAPI.cpp [hunk 1]
```diff
@@ -31,11 +31,18 @@ ClassDefine<EntityClass> EntityClassBuilder =
         .instanceProperty("blockPos", &EntityClass::getBlockPos)
         .instanceProperty("maxHealth", &EntityClass::getMaxHealth)
         .instanceProperty("health", &EntityClass::getHealth)
+        .instanceProperty("canFly", &EntityClass::getCanFly)
```
Comment (Jaciezyt) [inline ScriptEngine/API/EntityAPI.cpp]: remove `get` in `getCanXxx` or `getIsXxx` plz, this looks strange

File: docs/zh_CN/Development/GameAPI/Entity.md [hunk 2]
```diff
@@ -74,11 +74,17 @@
...
+| en.canFly    | 实体是否可以飞行        | `Boolean`        |
 | en.inAir     | 实体当前是否悬空   | `Boolean`        |
 | en.inWater   | 实体当前是否在水中 | `Boolean`        |
+| en.inLava     | 实体是否在熔岩中   | `Boolean`        |
+| en.inRain   | 实体是否在雨中 | `Boolean`        |
+| en.inSnow   | 实体是否在雪中 | `Boolean`        |
 | en.speed     | 实体当前速度       | `Float`          |
 | en.direction | 实体当前朝向       | `DirectionAngle` |
 | en.uniqueId  | 实体唯一标识符     | `String`         |
+| en.isTrading | 实体在交易 | `Boolean`    |
```
Comment (Jaciezyt) [inline docs/zh_CN/Development/GameAPI/Entity.md]:
```suggestion
| en.isTrading | 实体是否在交易 | `Boolean`    |
```

File: docs/zh_CN/Development/GameAPI/Entity.md [hunk 3]
```diff
@@ -74,11 +74,17 @@
...
+| en.isRiding | 实体正在骑行 | `Boolean`    |
```
Comment (Jaciezyt) [inline docs/zh_CN/Development/GameAPI/Entity.md]:
```suggestion
| en.isRiding | 实体是否在骑行 | `Boolean`    |
```

File: ScriptEngine/API/EntityAPI.cpp [hunk 4]
```diff
@@ -52,6 +59,9 @@ ClassDefine<EntityClass> EntityClassBuilder =
         .instanceFunction("refreshItems", &EntityClass::refreshItems)
         .instanceFunction("setNbt", &EntityClass::setNbt)
         .instanceFunction("getNbt", &EntityClass::getNbt)
+        .instanceFunction("getNameTag", &EntityClass::getNameTag)
```
Comment (Jaciezyt) [inline ScriptEngine/API/EntityAPI.cpp]: make `getNameTag` a property is beter

**Review by Jaciezyt — CHANGES_REQUESTED** [review]

File: ScriptEngine/API/EntityAPI.cpp (hunk 4 above)
Comment (ghost) [reply ScriptEngine/API/EntityAPI.cpp]: > make `getNameTag` a property is beter

**Changed!**

File: ScriptEngine/API/EntityAPI.cpp (hunk 1 above)
Comment (ghost) [reply ScriptEngine/API/EntityAPI.cpp]: > remove `get` in `getCanXxx` or `getIsXxx` plz, this looks strange

**Changed!**

**Pull request closed without merging**